k2/csrc/ragged_ops.cu

```diff
@@ -223,6 +223,122 @@ RaggedShape RaggedShape3(Array1<int32_t> *row_splits1,
   return RaggedShape(axes);
 }
 
+
+RaggedShape RaggedShape4(Array1<int32_t> *row_splits1,
+                         Array1<int32_t> *row_ids1, int32_t cached_tot_size1,
+                         Array1<int32_t> *row_splits2,
+                         Array1<int32_t> *row_ids2, int32_t cached_tot_size2,
+                         Array1<int32_t> *row_splits3,
+                         Array1<int32_t> *row_ids3, int32_t cached_tot_size3) {
+  NVTX_RANGE(__func__);
+  K2_CHECK(row_splits1 != nullptr || row_ids1 != nullptr)
+      << "At least one of row_splits1 and row_ids1 must be defined";
+  K2_CHECK(row_splits2 != nullptr || row_ids2 != nullptr)
+      << "At least one of row_splits2 and row_ids2 must be defined";
+  K2_CHECK(row_splits3 != nullptr || row_ids3 != nullptr)
+      << "At least one of row_splits3 and row_ids3 must be defined";
+
+
+  ContextPtr ctx = ::GetContext(row_splits1, row_ids1);
```

Comment:
`GetContext` is a variadic function.

```cpp
ContextPtr ctx = ::GetContext(row_splits1, row_ids1, row_splits2, row_ids2, row_splits3, row_ids3);
```
is simpler.

Reply:
mm, `::GetContext` is an internal function https://github.com/k2-fsa/k2/blob/ddd30a16d0c0fe28d82773737a939afca0b4e285/k2/csrc/ragged_ops.cu#L34-L35 which will check if the input is null or not, this is different from `k2::GetContext`

So I think there's no need to change.

Reply:
ok